source/crud/crud.rst

```diff
@@ -218,6 +218,10 @@ Read
      * If specified, drivers SHOULD apply this option to both the original aggregate command and subsequent
      * getMore operations on the cursor.
      *
+     * Drivers MUST NOT specify a batchSize of zero in an aggregate command that includes an $out stage, as
+     * that will prevent the pipeline from executing. Drivers SHOULD leave the cursor.batchSize command option
+     * unset in an aggregate command that includes an $out stage.
+     *
```

Comment:
I considered adding another paragraph with:

> Drivers that setup a cursor to be executed upon iteration against the $out collection MAY apply the batchSize option to that cursor, as that has no impact on the execution of the pipeline.

But that's outside the scope of the SPEC ticket. I also don't have any experience with drivers that do return cursors for `$out` pipelines and have no idea if they use batchSize in that additional capacity. I was careful to only refer to the initial `aggregate` command in the PR text (without any impact on other users of batchSize), so hopefully we can leave it at that.

Reply:
Ah, right, I forgot about how the spec gives drivers leeway on returning a cursor after a `$out` aggregation

Reply:
@jyemin - I believe the Java and/or .NET drivers return cursors that work with $out?

Reply:
In Java, Collection.aggregate returns an AggregateIterable.  AggregateIterable has a toCollection method that returns void, which throws if the pipeline doesn't terminate with $out.  It also has an iterator method that returns a MongoCursor, which for a $out aggregation as a cursor over the collection that you $out-ed to.

Reply:
I think in the latter case the batchSize is ignored for the aggregate, and used for establishing the cursor.

Reply:
> I think in the latter case the batchSize is ignored for the aggregate, and used for establishing the cursor.

@jyemin: Are you comfortable with the language in the PR? It doesn't sound as if it conflicts with Java's current behavior.

Reply:
Yeah, seems fine.